File: src/BetPlay.Domain/Fixture/Event.cs
using BetPlay.ApiSport.Dto.Fixture.Events;

namespace BetPlay.Domain.Fixture;

public class Event : Entity
{
    public Event()
    {
    }

    public Event(int fixtureId, EventApiDto eventApi, Team.Team team)
    {
        FixtureId = fixtureId;
        Elapsed = eventApi.Time.Elapsed;
        Extra = eventApi.Time.Extra;
        Team = team;
        PlayerId = eventApi.Player.Id;
        PlayerName = eventApi.Player.Name;
        AssistId = eventApi.Assist.Id;
        AssistName = eventApi.Assist.Name;
        Type = eventApi.Type;
        Detail = eventApi.Detail;
        Comments = eventApi.Comments;
    }

    public int FixtureId { get; set; }
    public int Elapsed { get; set; }
    public int? Extra { get; set; }
    public Team.Team Team { get; set; } = default!;
    public int? PlayerId { get; set; }
    public string? PlayerName { get; set; } = default!;
    public int? AssistId { get; set; }
    public string? AssistName { get; set; } = default!;
    public string Type { get; set; } = default!;
    public string Detail { get; set; } = default!;
    public string? Comments { get; set; } = default!;
}

File: betplay-web/types/fixture-details/event.ts
export type EventType = {
  time: TimeApiDto;
  team: TeamApiDto;
  player: PlayerApiDto;
  assist: AssistApiDto;
  type: string;
  detail: string;
  comments: string | null;
};

export type TimeApiDto = {
  elapsed: number;
  extra: number | null;
};

export type TeamApiDto = {
  id: number;
  name: string;
  logo: string;
};

export type PlayerApiDto = {
  id: number | null;
  name: string | null;
};

export type AssistApiDto = {
  id: number | null;
  name: string | null;
};
